web/src/components/graph/areagraph.js

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+} from 'react';
+import { scaleTime, scaleLinear } from 'd3-scale';
+import { first, last } from 'lodash';
+import moment from 'moment';
+
+import AreaGraphGradients from './areagraphgradients';
+import AreaGraphLayers from './areagraphlayers';
+import GraphBackground from './graphbackground';
+import GraphHoverLine from './graphhoverline';
+import ValueAxis from './valueaxis';
+import TimeAxis from './timeaxis';
+
+const X_AXIS_HEIGHT = 20;
+const Y_AXIS_WIDTH = 35;
+const Y_AXIS_PADDING = 4;
+
+const getTimeScale = (containerWidth, datetimes, currentTime) => scaleTime()
+  .domain([first(datetimes), currentTime ? moment(currentTime).toDate() : last(datetimes)])
+  .range([0, containerWidth]);
+
+const getValueScale = (containerHeight, maxTotalValue) => scaleLinear()
+  .domain([0, maxTotalValue * 1.1])
+  .range([containerHeight, Y_AXIS_PADDING]);
+
+const AreaGraph = React.memo(({
+  id,
+  layers,
+  currentTime,
+  maxTotalValue,
+  valueAxisLabel,
+  datetimes,
+  fillSelector,
+  mouseMoveHandler,
+  mouseOutHandler,
+  layerMouseMoveHandler,
+  layerMouseOutHandler,
+  selectedTimeIndex,
+  selectedLayerIndex,
+  gradientIdPrefix,
+  gradientLayers,
+  gradientStopColorSelector,
+  isMobile,
+}) => {
```

Comment:
We plan to re-use this component quite a lot.
Therefore, I'd like to spend a bit of time here making sure all props are well defined.
I'll add some comments on each prop